framework_crates/bones_ecs/src/entities.rs

```diff
@@ -286,6 +351,31 @@ macro_rules! impl_query {
                 )*
             }
 
+            #[allow(non_snake_case)]
+            fn get_single_with_bitset(
+                self,
+                bitset: Rc<BitSetVec>,
+            ) -> Result<<Self::Iter as Iterator>::Item, QuerySingleError> {
+                let (
+                    $(
+                        $args,
+                    )*
+                ) = self;
+                let mut query = MultiQueryIter {
+                    data: (
+                        $(
+                            $args.iter_with_bitset(bitset.clone()),
+                        )*
+                    )
+                };
+                let first = query.next();
+                let has_second = query.next().is_some();
+                match (first, has_second) {
+                    (Some(items), false) => Ok(items),
+                    _ => Err(QuerySingleError::MultipleEntities),
```

Comment:
Reading this I think that if first is None, this should return `QuerySingleError::NoEntities`.